File: Java/프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        int plus = (brown + 4) / 2;
        int mul = brown + yellow;
        
        for (int x=1; x<plus; x++) {
            int y = plus - x;
            if (x * y == mul) {
                answer[0] = y;
                answer[1] = x;
                return answer;
            }
        }
        return answer;
    }
}

File: JavaScript/프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    const plus = (brown + 4) / 2;
    const mul = brown + yellow;
    
    for (let x=1; x<plus; x++) {
        let y = plus - x;
        if (x * y == mul) return [y, x];
    }
}